interpretation/deepseismic_interpretation/azureml_pipelines/README.md

```diff
@@ -0,0 +1,212 @@
+    - If your script has local dependencies (i.e., is importing from another script) make sure that all dependencies fall underneath the source_directory
+2) You have configured your config file to specify the steps needed (see the section below "Configuring a Pipeline" for guidance)
+
+Note: the following arguments are automatically added to any script steps by AzureML:
+```--input_data``` and ```--output``` (if output is specified in the pipeline_config.json)
+Make sure to add these arguments in your scripts like so:
+```python
+parser.add_argument('--input_data', type=str, help='path to preprocessed data')
+parser.add_argument('--output', type=str, help='output from training')
+```
+```input_data``` is the absolute path to the input_datareference_path on the blob you specified.
+  
+# Configuring a Pipeline
+  
+## Train Pipeline
+Define parameters for the run in a config file. See an example [here](pipeline_config.json)
+```json
+{
+    "step1":
+    {
+        "type": "<type of step. Supported types include PythonScriptStep and MpiStep>",
+        "name": "<name in AzureML for this step>",
+        "script": "<path to script for this step>",
+        "output": "<name of the output in AzureML for this step - optional>",
+        "input_datareference_path": "<path on the data reference for the input data - optional>",
+        "input_datareference_name": "<name of the data reference in AzureML where the input data lives - optional>",
+        "input_dataset_name": "<name of the datastore in AzureML - optional>",
+        "source_directory": "<source directory containing the files for this step>",
+        "arguments": "<arguments to pass to the script - optional>",
+        "requirements": "<path to the requirements.txt file for the step - optional>",
+        "node_count": "<number of nodes to run the script on - optional>",
+        "processes_per_node": "<number of processes to run on each node - optional>",
+        "base_image": "<name of an image registered on dockerhub that you want to use as your base image"
+    },
+  
+    "step2":
+    {
+        .
+        .
+        .
+    }
+}
+```
+  
+## Kicking off a Pipeline
+In order to kick off a pipeline, you will need to use the AzureCLI to login to the subscription where your workspace resides:
+```bash
+az login
+az account set -s <subscription id>
+```
+Kick off the training pipeline defined in your config via your python environment of choice. The code will look like this:
+```python
+from src.azml.train_pipeline.train_pipeline import TrainPipeline
+  
+orchestrator = TrainPipeline("<path to your config file>")
+orchestrator.construct_pipeline()
+run = orchestrator.run_pipeline(experiment_name="DEV-train-pipeline")
+```
+See an example in [dev/kickoff_train_pipeline.py](dev/kickoff_train_pipeline.py)
+
+If this fails due to access to the Azure ML subscription, you may be able to connect by using a workaround:
```

Comment:
Perhaps add a Troubleshooting section? If we expect this code to fail then this doesn't instill a lot of confidence for the end user ;-)

Reply:
Added a Troubleshooting section in this commit: https://github.com/microsoft/seismic-deeplearning/pull/195/commits/19ff58f2b6e2b93f0e9c1abc6dcb4526fee25cb0